### ISSUE TITLE
Configuration test declares success even when errors are present

### ISSUE THREAD

**Issue by mike-somethingnooneelsehasusedyet** [fluent/fluent-bit#3162]
## Bug Report

**Describe the bug**
Using --dry-run to test configuration does not behave in a manner which allows easy automated detection of errors in configuration.

**To Reproduce**

```
[root@notneo:test:~]$ tail -1 /etc/td-agent-bit/td-agent-bit.conf 
NONSENSE NONSENSE NONSENSE
[root@notneo:test:~]$ /opt/td-agent-bit/bin/td-agent-bit --dry-run -c /etc/td-agent-bit/td-agent-bit.conf
Fluent Bit v1.7.1
* Copyright (C) 2019-2021 The Fluent Bit Authors
* Copyright (C) 2015-2018 Treasure Data
* Fluent Bit is a CNCF sub-project under the umbrella of Fluentd
* https://fluentbit.io

[2021/03/04 11:20:57] [  Error] File /etc/td-agent-bit/td-agent-bit.conf
[2021/03/04 11:20:57] [  Error] Error in line 53: Invalid indentation level
configuration test is successful
[root@notneo:test:~]$ 
[root@notneo:test:~]$ echo $?
0
[root@notneo:test:~]$ 
```

**Expected behavior**
"configuration test is successful" is displayed only when no errors in configuration have been detected.
exit status is `0` only when no errors in configuration have been detected.

**Your Environment**
```
[root@notneo:test:~]$ rpm -qf /opt/td-agent-bit/bin/td-agent-bit
td-agent-bit-1.7.1-1.x86_64
[root@notneo:test:~]$ cat /etc/os-release 
NAME="Oracle Linux Server"
VERSION="8.3"
ID="ol"
ID_LIKE="fedora"
VARIANT="Server"
VARIANT_ID="server"
VERSION_ID="8.3"
PLATFORM_ID="platform:el8"
PRETTY_NAME="Oracle Linux Server 8.3"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:oracle:linux:8:3:server"
HOME_URL="https://linux.oracle.com/"
BUG_REPORT_URL="https://bugzilla.oracle.com/"

ORACLE_BUGZILLA_PRODUCT="Oracle Linux 8"
ORACLE_BUGZILLA_PRODUCT_VERSION=8.3
ORACLE_SUPPORT_PRODUCT="Oracle Linux"
ORACLE_SUPPORT_PRODUCT_VERSION=8.3
[root@notneo:test:~]$ 

```
**Additional context**
I want my Ansible playbooks to be able to call the configuration test and the playbook run to stop if there's an error in the configuration. An exit status of `0` even when configuration is bad makes that far harder than it ought to be. This is an example of desirable behaviour: 
```
[root@notneo:test:~]$ apachectl configtest &>/dev/null || echo "stop your config is bad"
[root@notneo:test:~]$ echo "nonsense" >> /etc/httpd/conf.d/ssl.conf 
[root@notneo:test:~]$ apachectl configtest &>/dev/null || echo "stop your config is bad"
stop your config is bad
[root@notneo:test:~]$ 
```
This Ansible task causes a playbook run to be aborted in the event that bad configuration has been put in place
```
- name: apache config test
  shell: /usr/sbin/apachectl configtest
  changed_when: false
```

This task should achieve the same but it doesn't because the command exits with status `0` when the configuration is bad.
```
- name: td-agent-bit config test
  shell: /opt/td-agent-bit/bin/td-agent-bit --dry-run -c /etc/td-agent-bit/td-agent-bit.conf
  changed_when: false
```

**Comment by yongtang** [comment]
@mike-somethingnooneelsehasusedyet With the latest master the return code is `1`:
```
$ cat fluent-bit.conf 
NONSENSE NONSENSE NONSENSE
ubuntu@ip-172-31-87-192:~/v/fluent-bit/build$ bin/fluent-bit --dry-run -c fluent-bit.conf 
Fluent Bit v1.8.0
* Copyright (C) 2019-2021 The Fluent Bit Authors
* Copyright (C) 2015-2018 Treasure Data
* Fluent Bit is a CNCF sub-project under the umbrella of Fluentd
* https://fluentbit.io

[2021/03/05 17:56:55] [  Error] Error: there are not sections available on fluent-bit.conf!
Error: No Input(s) have been defined. Aborting

$ echo $?
1
```

Can you try with the master build and see if the issue still persists?

**Comment by nokute78** [comment]
Hmm, I can reproduce using this configuration file with latest master https://github.com/fluent/fluent-bit/tree/f58b7ca0f03672417374bf5a67bda67261f642e6

```
[INPUT]
    Name cpu

[OUTPUT]
    Name stdout

NONSENSE NONSENSE NONSENSE
```

output: 
```
taka@locals:~/git/fluent-bit/build/3162$ ../bin/fluent-bit -c a.conf --dry-run
Fluent Bit v1.8.0
* Copyright (C) 2019-2021 The Fluent Bit Authors
* Copyright (C) 2015-2018 Treasure Data
* Fluent Bit is a CNCF sub-project under the umbrella of Fluentd
* https://fluentbit.io

[2021/03/06 07:39:50] [  Error] File a.conf
[2021/03/06 07:39:50] [  Error] Error in line 7: Invalid indentation level
configuration test is successful
taka@locals:~/git/fluent-bit/build/3162$ echo $?
0
```

**Comment by yongtang** [comment]
@nokute78 Added a PR #3187 for the fix.

**Comment by mike-somethingnooneelsehasusedyet** [comment]
Issue still present in latest version, 1.7.3.
```
[root@notneo:test:~]$ /opt/td-agent-bit/bin/td-agent-bit --dry-run -c /etc/td-agent-bit/td-agent-bit.conf
Fluent Bit v1.7.3
* Copyright (C) 2019-2021 The Fluent Bit Authors
* Copyright (C) 2015-2018 Treasure Data
* Fluent Bit is a CNCF sub-project under the umbrella of Fluentd
* https://fluentbit.io

[2021/04/12 15:03:34] [  Error] File /etc/td-agent-bit/td-agent-bit.conf
[2021/04/12 15:03:34] [  Error] Error in line 53: Invalid indentation level
configuration test is successful
[root@notneo:test:~]$ echo $?
0
[root@notneo:test:~]$ 

```

**Comment by nokute78** [comment]
@mike-somethingnooneelsehasusedyet Hmm, The patch is not merged yet.

FYI: Fluent-bit uses other OSS "monkey" as utility library.
@yongtang sent a patch to monkey and it was fixed.
However Fluent-bit is not merged fixed monkey.

@edsiper When monkey will be merged ?

**Comment by mike-somethingnooneelsehasusedyet** [comment]
Seems to be fixed in 1.7.4 
```
[root@mike-test4:test:~]$ tail -1 /etc/td-agent-bit/td-agent-bit.conf
NONSENSE
[root@mike-test4:test:~]$ /opt/td-agent-bit/bin/td-agent-bit --dry-run -c /etc/td-agent-bit/td-agent-bit.conf
Fluent Bit v1.7.4
* Copyright (C) 2019-2021 The Fluent Bit Authors
* Copyright (C) 2015-2018 Treasure Data
* Fluent Bit is a CNCF sub-project under the umbrella of Fluentd
* https://fluentbit.io

[2021/05/17 14:52:29] [  Error] File /etc/td-agent-bit/td-agent-bit.conf
[2021/05/17 14:52:29] [  Error] Error in line 53: Invalid indentation level
[2021/05/17 14:52:29] [  Error] File /etc/td-agent-bit/td-agent-bit.conf
[2021/05/17 14:52:29] [  Error] Error in line 53: Each key must have a value
Error: Configuration file contains errors. Aborting

[root@mike-test4:test:~]$ echo $?
1
[root@mike-test4:test:~]$ 

```
